packages/sw-lib/src/lib/router.js

```diff
@@ -112,14 +121,23 @@ class Router extends SWRoutingRouter {
    * @param {Array<RegExp>} [options.blacklist] Defaults to an empty blacklist.
    * @param {Array<RegExp>} [options.whitelist] Defaults to `[/./]`, which will
    *        match all request URLs.
+   * @param {String} [options.cacheName] The name of the cache which contains
+   *        the cached response for `url`. Defaults to the name of the cache
+   *        used by cacheRevisionedAssets().
    */
   registerNavigationRoute(url, options = {}) {
     if (typeof url !== 'string') {
       throw ErrorFactory.createError('navigation-route-url-string');
     }
 
+    // Allow folks to explicitly pass in a null/undefined cacheName option if
+    // they want that behavior.
+    const cacheName = 'cacheName' in options ?
```

Comment:
Passing in `null`/`undefined` will default to the `caches.match()` behavior that was in place prior to this PR, i.e. return the first match for the URL, regardless of which cache it's in. Not necessarily the ideal use case (hence this PR), but I figured I'd provide a backdoor in case people really wanted that behavior?